packages/browser/src/sdk.ts

```diff
@@ -170,32 +125,51 @@ declare const __SENTRY_RELEASE__: string | undefined;
  * @see {@link BrowserOptions} for documentation on configuration options.
  */
 export function init(browserOptions: BrowserOptions = {}): Client | undefined {
+  // Note: If we call `initWithDefaultIntegrations()` here, webpack seems unable to tree-shake the DEBUG_BUILD usage inside of it
+  // So we duplicate the logic here like this to ensure maximum saved bytes
   const options = applyDefaultOptions(browserOptions);
+  const defaultIntegrations = getDefaultIntegrations(browserOptions);
 
-  if (!options.skipBrowserExtensionCheck && shouldShowBrowserExtensionError()) {
-    if (DEBUG_BUILD) {
-      consoleSandbox(() => {
-        // eslint-disable-next-line no-console
-        console.error(
-          '[Sentry] You cannot run Sentry this way in a browser extension, check: https://docs.sentry.io/platforms/javascript/best-practices/browser-extensions/',
-        );
-      });
-    }
+  const showBrowserExtensionError = !options.skipBrowserExtensionCheck && shouldShowBrowserExtensionError();
+
+  if (DEBUG_BUILD) {
+    logBrowserEnvironmentWarnings({
+      browserExtension: showBrowserExtensionError,
+      fetch: !supportsFetch(),
+    });
+  }
+
+  if (showBrowserExtensionError) {
     return;
   }
 
-  if (DEBUG_BUILD && !supportsFetch()) {
-    logger.warn(
-      'No Fetch API detected. The Sentry SDK requires a Fetch API compatible environment to send events. Please add a Fetch API polyfill.',
-    );
+  const clientOptions = getClientOptions(options, defaultIntegrations);
+  return initAndBind(BrowserClient, clientOptions);
+}
+
+/**
+ * Initialize a browser client with the provided options and default integrations getter function.
+ */
```

Comment:
l: Based on your PR description:

```suggestion
/**
 * Initialize a browser client with the provided options and default integrations getter function.
 *
 * @hidden
 * @internal - This is only intended for internal use. Use at your own risk!
 */
```